docs/administrators/policy_cookbook.md

```diff
@@ -568,3 +568,226 @@ pep_api_data_obj_close_post(*INSTANCE_NAME, *COMM, *OPENED_DATA_OBJ_INPUT)
+```javascript
+{
+    "replicas": [
+        {
+            "before": {
+                "data_id": <string>,
+                "coll_id": <string>,
+                "data_repl_num": <string>,
+                "data_version": <string>,
+                "data_type_name": <string>,
+                "data_size": <string>,
+                "data_path": <string>,
+                "data_owner_name": <string>,
+                "data_owner_zone": <string>,
+                "data_is_dirty": <string>,
+                "data_status": <string>,
+                "data_checksum": <string>,
+                "data_expiry_ts": <string>,
+                "data_map_id": <string>,
+                "data_mode": <string>,
+                "r_comment": <string>,
+                "create_ts": <string>,
+                "modify_ts": <string>,
+                "resc_id": <string>
+            },
+            "after": {
+                "data_id": <string>,
+                "coll_id": <string>,
+                "data_repl_num": <string>,
+                "data_version": <string>,
+                "data_type_name": <string>,
+                "data_size": <string>,
+                "data_path": <string>,
+                "data_owner_name": <string>,
+                "data_owner_zone": <string>,
+                "data_is_dirty": <string>,
+                "data_status": <string>,
+                "data_checksum": <string>,
+                "data_expiry_ts": <string>,
+                "data_map_id": <string>,
+                "data_mode": <string>,
+                "r_comment": <string>,
+                "create_ts": <string>,
+                "modify_ts": <string>,
+                "resc_id": <string>
+            },
+            "file_modified": {
+                <string>: <string>,
+                ...
+            }
+        },
+        ...
+    ]
+}
+```
+Each entry in the list of replicas is a "before" and "after" view of the row in `R_DATA_MAIN` representing that replica, complete with the column names. In order to check that the replica has a checksum, we want to find the replica which was modified in the list, and ensure that the "after" entry's "data_checksum" entry has a value. If not, we will calculate a checksum and register it in the catalog.
+
+The `data_object_finalize` database PEP is invoked twice per data transfer: once on open in order to lock the data object, and once on close in order to finalize the system metadata. We are only interested in the invocation on close. You will find in the implementation checks which can detect which is being invoked. You might consider abstracting this into a separate function for more readable code.
```

Comment:
Can you explain what this line means?

> You will find in the implementation checks which can detect which is being invoked.

Reply:
It is saying that the example implementation has checks to detect whether the PEP was invoked by an open or close call. I've updated the wording to be clearer.